app/services/config_data_sources/json_project_data_source.rb

```diff
@@ -119,6 +119,17 @@ def projects
 
         saved_projects = JSON.parse(File.read(path)).map do |project_json|
           project = Project.from_json!(project_json)
+
+          # Now load the project specific files here
+          # that are in
+          #
+          #   `projects/[project_id]/[file].json
+          #
+          # more information on https://github.com/fastlane/ci/issues/643
+          project_specific_path = File.join(git_repo.file_path("projects"), project.id)
+          environment_variable_data_source = JSONEnvironmentDataSource.create(project_specific_path)
+          project.environment_variables = environment_variable_data_source.environment_variables
```

Comment:
Would love @taquitos' input on this - currently I'm creating a `JSONEnvironmentDataSource` for each project within the `json_project_data_source`, which would allow us to have a different data source for ENV variables (secrets) depending on the project. I wasn't sure if I should access the `EnvironmentVariableService` here or the ``JSONEnvironmentDataSource` directly, what's the approach here?

Reply:
This is a great idea! I think datasource to datasource communication is best here. The thinking is that you should be able to bundle the datasources together into a package with no dependences on the services. Thinking of it as a layer.